### ISSUE TITLE
[ONNX] Support Sum operator

### ISSUE THREAD

**Issue by arogozhnikov** [dmlc/nnvm#375]
Hi, I've tried to follow an example of importing from ONNX. I've used ShuffleNet from onnx/models.

```python
import nnvm
import tvm
import onnx
import numpy as np

onnx_graph = onnx.load('./onnx-android/data/shufflenet/model.pb')
sym, params = nnvm.frontend.from_onnx(onnx_graph)
```

And result is:

```
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-16-c70e34408bea> in <module>()
      5 
      6 onnx_graph = onnx.load('./onnx-android/data/shufflenet/model.pb')
----> 7 sym, params = nnvm.frontend.from_onnx(onnx_graph)

~/projects/nnvm/python/nnvm/frontend/onnx.py in from_onnx(graph)
    526         # it's a ModelProto wrapper
    527         graph = graph.graph
--> 528     sym, params = g.from_onnx(graph)
    529     return sym, params

~/projects/nnvm/python/nnvm/frontend/onnx.py in from_onnx(self, graph)
    402             attr = self._parse_attr(node.attribute)
    403             inputs = [self._nodes[self._renames.get(i, i)] for i in node.input]
--> 404             op = self._convert_operator(op_name, inputs, attr)
    405             node_output = self._fix_outputs(op_name, node.output)
    406             assert len(node_output) == len(op.list_output_names()), (

~/projects/nnvm/python/nnvm/frontend/onnx.py in _convert_operator(self, op_name, inputs, attrs, identity_list, convert_map)
    487             sym = convert_map[op_name](inputs, attrs, self._params)
    488         else:
--> 489             raise NotImplementedError("Operator {} not implemented.".format(op_name))
    490         return sym
    491 

NotImplementedError: Operator Sum not implemented.
```
Is it a bug or indeed sum was not implemented?

**Comment by tqchen** [comment]
I think it is a problem of front end, sum is implemented in nnvm core tensor, depending on whether it is reduction or element wise add, we can follow up the conversion

**Comment by tqchen** [comment]
contribution is welcomed

**Comment by srkreddy1238** [comment]
This onnx model has multiple issues to solve.

1: Sum operator (Time being we could use normal topi add as all has only 2 inputs)
2: Dense : Dense data input require flatten (1x544x1x1) - Need to workout some solution to Handle this in frontend or in InferShape & topi.
3: Convolution inputs has arbitrary groups : Not supported yet. 

Siva

**Comment by tqchen** [comment]
for 3, we already support efficient depthwise, and we can specialize to ngroup equals number of channels and ngroup=1 to different operators, which covers almost all cases. Can add more generic group support(which can be less efficient) later in topi

**Comment by tqchen** [comment]
@srkreddy1238 interested in resolving the sum operator issue first?

**Comment by srkreddy1238** [comment]
1: Patch for Sum operator @ https://github.com/dmlc/nnvm/pull/386

SRK

**Comment by tqchen** [comment]
fixed by #386
